### PR TITLE
feat(primarynav): adding AIOutlineStar icon and Star us on Github text to PrimaryNav

### PR DESCRIPTION
**Pull request by AlexVCS**
## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description

This PR is working to add the icon and star us on GitHub text to the nav 
## Related Tickets & Documents

Feature #289 
## Mobile & Desktop Screenshots/Recordings

Screen Shot 2022-08-05 at 1 41 43 PM

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [x] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Comment by 0-vortex** [comment]
`npm run format` is your friend! Should fix most of the red dots 🍕

**Comment by bdougie** [comment]
checking in @AlexVCS to see if you have everything you need.

**Comment by AlexVCS** [comment]
> checking in @AlexVCS to see if you have everything you need.
Screen Shot 2022-08-08 at 8 22 56 PM

Hey Brian, as you can see in the screenshot, it looks decent on desktop. Needs refinement for mobile. A family funeral has come up all of a sudden, so I hope I can get feedback on what I've got and work more on this Thursday or Friday (my apologies for the delay).

**Comment by bdougie** [comment]
No worries @AlexVCS. A few things. 

- move the link to the right next to the submit
- change to `test-sm`

**Comment by AlexVCS** [comment]
Thanks for the feedback Brian! I've applied what you said, let me know if I need to make any changes!

Screen Shot 2022-08-12 at 1 02 46 PM

**Comment by bdougie** [comment]
Mobile view is off image

**Comment by AlexVCS** [comment]
Gotcha, should be fixed now!
Screen Shot 2022-08-12 at 3 02 20 PM

File: src/components/PrimaryNav.tsx [hunk 1]
```diff
@@ -123,15 +124,29 @@ const PrimaryNav = (): JSX.Element => {
...
+                target="_blank"
+              >
+                <AiOutlineStar className="justify-self-start inline-block mr-[10px]" />
+
+                <span className="text-sm">Star us on GitHub</span>
+              </a>
+            </div>
+
+            <button
+              className="bg-osOrange w-[64px] h-[24px]  rounded-[6px] px-[12px] py-[2px] text-xs font-semibold text-white md:ml-[20px]"
```
Comment (bdougie) [inline src/components/PrimaryNav.tsx:141]: This pattern of hard coding px is an anti-pattern of tailwind. https://github.com/open-sauced/hot/issues/267

**Review by bdougie — APPROVED** [review]
LGTM. The hardcoded px in the class is a carry over, but linking the issue and noting it will need to be fixed. https://github.com/open-sauced/hot/issues/267

**Comment by bdougie** [comment]
Just realized this does not show when logged in, that would be ideal.

**Comment by AlexVCS** [comment]
Not the most DRY, but I have it showing up when logged in now too. I'm using another tailwind anti-pattern, I was looking up a few things about how to avoid this, but am unsure and I know that issue #267 is still being worked on. If you've got any feedback on how to get away from that now, or at all please let me know!

Screen Shot 2022-08-14 at 12 20 47 PM

**Comment by bdougie** [comment]
Only because I saw you are looking for work on your Twitter, I will need to push back once more and request that this div be placed into a new reusable component. You put it in the same file, but duplicating code is not going to impress hiring managers. 

const StarTheRepo = (): JSX.Element => (
   <div className="hidden md:flex items-center text-osGrey">
     <a
        href="https://github.com/open-sauced/hot"
        rel="noreferrer"
        target="_blank"
      >
        <AiOutlineStar className="inline-block mr-[10px]" />

         <span className="text-sm mr-[10px]">Star us on GitHub</span>
     </a>
  </div>
)

**Comment by AlexVCS** [comment]
Thanks Brian! I made it a reusable component.
Screen Shot 2022-08-15 at 7 01 50 AM

**Comment by 0-vortex** [comment]
`hidden` makes it hidden on all screen while `md:flex` makes it flex visible on medium devices and up, we should apply a mobile-first approach to this if possible, rather than duplicate the element in 2 places for example the header (hidden md:flex) and footer (flex md:hidden) - this is an anti-pattern.

I propose:
- removing `hidden and md:hidden` classes in favour of simply `flex` while changing the font to `font-Inter`, the final class could look like this: flex `items-center text-osGrey font-Inter`
- changing line 21 from `<span className="text-sm mr-[10px]">Star us on GitHub</span>` to `<span className="text-md font-light mr-[10px]">Star us on GitHub</span>`
- move the closing `div` on line 61 to line 65 to encompass the `GiHamburgerMenu` and its parent
- making the mobile hamburger menu slightly bigger to be in line with the star link and opensauced logo, increasing the sizes from 20px to 24px, with the line 61 change this element should look like ```<div className="flex md:hidden w-[24px] h-[24px]">
                  <GiHamburgerMenu size={24} />
                </div>
              </div>
            </Menu.Button>```

Doing all the above should provide for a seamless experience on all platforms.

Sorry for the checkbox style changelist, when moving markup like child into parent containers, it makes proposing changes very verbose and I wanted for these to also make sense if you encounter similar problems in the future! 🍕

**Comment by AlexVCS** [comment]
> `hidden` makes it hidden on all screen while `md:flex` makes it flex visible on medium devices and up, we should apply a mobile-first approach to this if possible, rather than duplicate the element in 2 places for example the header (hidden md:flex) and footer (flex md:hidden) - this is an anti-pattern.
> 
> I propose:
> 
> * removing `hidden and md:hidden` classes in favour of simply `flex` while changing the font to `font-Inter`, the final class could look like this: flex `items-center text-osGrey font-Inter`
> * changing line 21 from `<span className="text-sm mr-[10px]">Star us on GitHub</span>` to `<span className="text-md font-light mr-[10px]">Star us on GitHub</span>`
> * move the closing `div` on line 61 to line 65 to encompass the `GiHamburgerMenu` and its parent
> * making the mobile hamburger menu slightly bigger to be in line with the star link and opensauced logo, increasing the sizes from 20px to 24px, with the line 61 change this element should look like
> 
> ```
>     <div className="flex md:hidden w-[24px] h-[24px]">
>       <GiHamburgerMenu size={24} />
>     </div>
>   </div>
> </Menu.Button>
> ```
> 
> Doing all the above should provide for a seamless experience on all platforms.
> 
> Sorry for the checkbox style changelist, when moving markup like child into parent containers, it makes proposing changes very verbose and I wanted for these to also make sense if you encounter similar problems in the future! 🍕

The button and icon appear on mobile with these changes. Brian previously mentioned hiding it for mobile so that's why I had them hidden on mobile. The spacing and alignment is OK when signed in, but not great when not signed in. I can work to optimize the spacing for mobile, is that what you'd like? I can make the sign in button bigger, and decrease the size of the star us on GitHub text.

I've included screenshots of the changes you recommended in mobile view.

Signed in:
Screen Shot 2022-08-15 at 9 59 09 AM

Not signed in:
Screen Shot 2022-08-15 at 9 59 56 AM

**Comment by 0-vortex** [comment]
You are right, my mobile screen is too big 😓

**Comment by bdougie** [comment]
Yeah, mobile doesn't work with all the words. An alternative would be only showing the star.


Another option is using a smaller font size + less words

Up to you and your creativity, but 'hidden sm:flex' works as a bare minimum

**Comment by AlexVCS** [comment]
Yeah, I've gone back to making it hidden on mobile and pushed up the change.

Screen Shot 2022-08-16 at 11 03 27 AM

**Review by bdougie — APPROVED** [review]
LGTM

**Pull request merged**